src/number.c

```diff
@@ -417,3 +418,28 @@ scm_p_remainder(ScmObj _n1, ScmObj _n2)
 
     return MAKE_INT(n1 % n2);
 }
+
+SCM_EXPORT ScmObj
+scm_p_expt(ScmObj _base, ScmObj _expo)
+{
+    scm_int_t base, expo, result;
+    DECLARE_FUNCTION("expt", procedure_fixed_2);
+
+    ENSURE_INT(_base);
+    ENSURE_INT(_expo);
+
+    base = SCM_INT_VALUE(_base);
+    expo = SCM_INT_VALUE(_expo);
+	
+    if (expo < 0) ERR(ERRMSG_NEGATIVE_EXP);
+
+    result = 1;
+    while (expo > 0) {
+	if (expo % 2 == 1)
+	    result *= base;
+	base *= base;
+	expo /= 2;
```

Comment:
Could you use spaces instead of tabs for indent?

Reply:
Sorry, fixed!